enzyme/Enzyme/MLIR/Implementations/ArithAutoDiffOpInterfaceImpl.cpp

```diff
@@ -75,11 +80,108 @@ struct AddFOpInterface
     return success();
   }
 };
+
+void addToGradient(Value oldGradient, Value addedGradient, OpBuilder &builder,
+                   MGradientUtilsReverse *gutils) {
+  Value gradient = addedGradient;
+  if (gutils->hasInvertPointer(oldGradient)) {
+    Value operandGradient = gutils->invertPointerM(oldGradient, builder);
+    gradient = builder.create<arith::AddFOp>(oldGradient.getLoc(),
```

Comment:
This may not work for all types

Reply:
I believe it should work for the same subset of types that are supported by the `arith` dialect (tensor, vector, and float), and it should only be called on those types. Would you prefer the use of the `createAddOp` interface method?

Reply:
yeah I would use the interface method.

Reply:
ah never mind this is already arith specifc